### ISSUE TITLE
Java 8 hangs (not 1.8.0_05 though)

### ISSUE THREAD

**Issue by SupaHam** [Microsoft/WSL#857]
Please use the following bug reporting template to help produce actionable and reproducible issues:

Java 8 hangs when typing `java -version`. I've tested this with a few installations.

I first tried the popular `ppa:webupd8team/java`:

```
$ sudo add-apt-repository ppa:webupd8team/java
$ sudo apt-get update
$ sudo apt-get install oracle-java8-installer
```

That installed successfully, but it hung when I typed `java -version`. Here's the [strace results](http://www.pastebucket.com/551797).

Then I went to check out if others were experiencing the same issue when I came across #49 specifically [this post](https://github.com/Microsoft/BashOnWindows/issues/49#issuecomment-217297195) of manual installation instructions for java 1.8.0_05.

I tried it out and it works just fine:

```
root@ALI:/home/supaham# java -version
java version "1.8.0_05"
Java(TM) SE Runtime Environment (build 1.8.0_05-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.5-b02, mixed mode)
```

However 1.8.0_05 has a few bugs and invalid compilation errors.

So I went to update to `1.8.0_102` but then it hung again. The instructions to install 102 were the same as the one in the post linked above, but the url replaced with:

```
wget --header "Cookie: oraclelicense=accept-securebackup-cookie" http://download.oracle.com/otn-pub/java/jdk/8u102-b14/jdk-8u102-linux-x64.tar.gz
```

Here's the [strace results](http://www.pastebucket.com/551809) of the `java -version` command on version `1.8.0_102`

Windows build number: Microsoft Windows [Version 10.0.14393]

No packages are required to install that I know of.

Edit: Although I can get `java -version` to work, I can't get a Minecraft instance to run without it hanging after displaying a message saying "Loading Libraries", usually on the same computer in normal Windows this step takes at most 5 seconds. So far it's been over two minutes and it's still stuck on that step with a very similar strace result as far as I can see.

**Comment by russalex** [comment]
Java has some known issues on the Anniversary Update.  The strace you provide looks very similar to some of the issues I have seen.  I believe that specific issue should be in the fix for #746.

**Comment by sunilmut** [comment]
I was unable to repro this in the 14971 build after following the repro `$ sudo add-apt-repository ppa:webupd8team/java $ sudo apt-get update $ sudo apt-get install oracle-java8-installer`

But, this is 1.8.0_111
![java8](https://cloud.githubusercontent.com/assets/16765049/20414622/2919cb4c-ace8-11e6-848a-092200c6ea76.png)
